crates/driver/src/domain/competition/mod.rs

```diff
@@ -328,6 +302,127 @@ impl Competition {
     }
 }
 
+/// Encode the solutions into settlements.
+async fn encode_solutions(
+    auction: Auction,
+    eth: Ethereum,
+    simulator: Simulator,
+    solutions: Vec<Solution>,
+) -> Vec<(solution::Id, Result<Settlement, solution::Error>)> {
+    let (sender, mut receiver) = tokio::sync::mpsc::unbounded_channel();
+    let auction = Arc::new(auction);
+    let eth = Arc::new(eth);
+    let simulator = Arc::new(simulator);
```

Comment:
I think we should probably eventually store all these important types in `Arc`s because they will get used in futures like this anyway.